### ISSUE TITLE
Semantic `GEOM` vs. `PART` sometimes mixed up, failing designator resolution (again)

### ISSUE THREAD

**Issue by fairlight1337** [cram2/cram_semantic_maps#6]
@gaya-, although I thought all the cases were covered in https://github.com/cram2/cram_semantic_maps/issues/5, apparently this is not the case:

While this works
```lisp
LTFNP> (top-level
         (with-designators ((loc :location `((:on "CounterTop")
                                             (:name "iai_kitchen_kitchen_island_counter_top"))))
           (cram-designators:reference loc)))
#<CL-TRANSFORMS-STAMPED:POSE-STAMPED 
   FRAME-ID: "map", STAMP: 0.0
   #<3D-VECTOR (-0.5999999046325684d0 1.0999999046325684d0 0.8509999997764826d0)>
   #<QUATERNION (0.0d0 0.0d0 0.0d0 1.0d0)>>
```
this does not
```lisp
LTFNP> (top-level
         (with-designators ((loc :location `((:on "CounterTop")
                                             (:name "iai_kitchen_kitchen_island_counter_top")))
                            (obj :object `((:at ,loc)))
                            (to-see :location `((:to :see)
                                                (:obj ,obj))))
           (cram-designators:reference to-see)))
WARNING:
   An error occurred while executing the lisp function `MERGE-COSTMAPS': `The value
                                                                            #<CRAM-SEMANTIC-MAP-UTILS:SEMANTIC-MAP-PART

                                                                              {100799C773}>
                                                                          is not of type
                                                                            CRAM-SEMANTIC-MAP-UTILS:SEMANTIC-MAP-GEOM.'
```
Somehow the differentiation between the two object types is not clean here. Can you look into this?

I put the full backtrace into a pastebin [here](http://pastebin.com/JmMHC7J7).

**Comment by fairlight1337** [comment]
This is kind of a blocker for multiple things; not being able to indirectly resolving designators (like (to see),  (to reach)) makes everything break that is supposed to happen mobile (like several perception runs and most importantly mobile manipulation).

So I'd like to nudge up priority on this one, @gaya-.

**Comment by gaya-** [comment]
Solved in fc55bb298776253e796ce972802cd1e90715e857.
